### PR TITLE
executor,store/mockstore: add a test to mock get tso fail

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Make our code more robust

### What is changed and how it works?

Add test

### Check List 

Tests 

 - Unit tes
 - No code

Related changes

release-2.0 and release 2.1 may fail on this test

 - Need to cherry-pick to the release branch

@winkyao  @crazycs520

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8566)

### CODE REVIEW

File: store/mockstore/mocktikv/pd.go [hunk 1]
```diff
@@ -98,6 +104,7 @@ func (c *pdClient) GetStore(ctx context.Context, storeID uint64) (*metapb.Store,
 
 func (c *pdClient) GetAllStores(ctx context.Context) ([]*metapb.Store, error) {
 	panic("unimplemented")
+	return nil, errors.New("for gofail")
```
Comment (crazycs520) [inline store/mockstore/mocktikv/pd.go]: redundant ?

Comment (tiancaiamao) [reply store/mockstore/mocktikv/pd.go]: Make golint happy...

**Comment by tiancaiamao** [comment]
PTAL @crazycs520 @winkyao

**Comment by tiancaiamao** [comment]
@lysu @jackysp

**Pull request merged**